packages/hash/frontend/src/blocks/page/BlockConfigMenu/BlockConfigMenu.tsx

```diff
@@ -0,0 +1,195 @@
+    blockSchema.configProperties?.includes(name),
+  );
+
+const resolvePropertySchema = (
+  $ref: string,
+  rootSchema: JsonSchema,
+): JsonSchema => {
+  if ($ref.startsWith("#/")) {
+    const deepObjectPath = $ref.split("/").slice(1).join(".");
+    return get(rootSchema, deepObjectPath);
+  }
+  throw new Error(`Resolution of external schema ${$ref} not yet implemented.`);
+};
+
+/**
+ * This is a temporary implementation of 'provide an input based on a property schema'
+ * We will need a more comprehensive implementation for the full entity editor
+ */
+const ConfigurationInput: VoidFunctionComponent<{
+  name: string;
+  onChange: (value: any) => void;
+  rootSchema: JsonSchema;
+  propertySchema: JsonSchema;
+  value?: JSONValue | null;
+}> = ({ name, onChange, propertySchema, rootSchema, value }) => {
+  const resolvedPropertySchema = propertySchema.$ref
+    ? resolvePropertySchema(propertySchema.$ref, rootSchema)
+    : propertySchema;
+  const { enum: enumList, format, type } = resolvedPropertySchema;
+
+  const updateProperty = (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => onChange(event.target.value);
+
+  const [draftValue, setDraftValue] = useState(value);
+
+  /**
+   * Keep track of the previous externally-provided value, in case the entity is updated while the menu is open.
+   */
+  const previousValue = useRef<JSONValue | undefined>(undefined);
+  useEffect(() => {
+    previousValue.current = value;
+  });
+
+  if (previousValue.current !== value && value !== draftValue) {
+    setDraftValue(value);
+  }
+
+  switch (type) {
+    case "boolean":
+      return (
+        <FormControlLabel
+          control={
+            <Checkbox
+              onChange={updateProperty}
+              checked={typeof value === "boolean" ? value : false}
+            />
+          }
+          label={name}
+        />
+      );
+
+    case "string":
+      if (format) {
+        // @todo validate string format - should have a reusable input that does this
+      }
+      return (
+        <TextField
+          label={name}
+          onBlur={enumList ? undefined : updateProperty}
+          onChange={(event) => {
+            setDraftValue(event.target.value);
+            if (enumList) {
+              updateProperty(event);
+            }
+          }}
+          select={!!enumList}
+          value={draftValue}
+          variant="outlined"
+        >
+          {(enumList ?? []).map((option: string) => (
+            <MenuItem key={option} value={option}>
+              {option}
+            </MenuItem>
+          ))}
+        </TextField>
+      );
+
+    case "number":
+      return (
+        <TextField
+          defaultValue={value ?? ""}
+          onBlur={(event) => onChange(event.target.value)}
+          onChange={(event) => setDraftValue(event.target.value)}
+          type="number"
+          value={draftValue}
+          variant="outlined"
```

Comment:
nit: there's no need to pass this because that's the default variant the component uses 
```suggestion
          variant="outlined"
```

Reply:
https://github.com/hashintel/hash/pull/490/commits/c17be4e9e3738af9d81cf5b52ce2816f77672426